### PR TITLE
PLT-1222 ADR about changing the slot to time conversions in the emulator and PAB

### PR DESCRIPTION
**Pull request by koslambrou**
* Added the ADR in the RTD

* Fixed the nix-shell cardano-node version which was out of date

Pre-submit checklist:
- Branch
    - [ ] Tests are provided (if possible)
    - [x] Commit sequence broadly makes sense
    - [x] Key commits have useful messages
    - [x] Formatting, PNG optimization, etc. are updated
    - [ ] Important changes are reflected in changelog.d of the affected packages
- PR
    - [x] Self-reviewed the diff
    - [x] Useful pull request description
    - [ ] Reference the ADR in the PR and reference the PR in the ADR (if revelant)
    - [x] Reviewer requested

### CODE REVIEW

File: doc/adr/0015-time-conversion.rst [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+--------
+
+* We will deprecate the ``Ledger.TimeSlot.SlotConfig`` type and all functions in the
+  ``Ledger.TimeSlot`` module using the ``SlotConfig``. The only viable functions are the ones that
+  convert between ``Data.Time`` types and plutus types (types related to ``TxInfo``).
+
+* We will copy the ``Ledger.TimeSlot`` module in the emulator (ideally rename it) and keep it as
+  an internal module. Any functions not used by the emulator will be removed.
+
+* We will move the ``Ledger.Params`` module inside the emulator as an internal module and modify the
```
Comment (sjoerdvisscher) [inline doc/adr/0015-time-conversion.rst:42]: Maybe we can just remove `SlotConfig` from `Params`. It doesn't need to be configurable anymore.

File: doc/adr/0015-time-conversion.rst [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+      ...
+
+* We will implement the emulator effect interpreter by simply using the ``SlotConfig`` for the conversions.
+
+* We will implement the PAB effect interpreter by using the local node. There are multiple steps to
+  implement this:
+
+  * At startup, the PAB will query the ``EraHistory`` from the local node and store it in its local
+    environment.
+  * We will implement the PAB interpreter by using the ``EraHistory`` alonside the consensus
```
Comment (kayvank) [inline doc/adr/0015-time-conversion.rst:72]:
```suggestion
  * We will implement the PAB interpreter by using the ``EraHistory`` alongside the consensus
```

File: doc/adr/0015-time-conversion.rst [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+  convert between ``Data.Time`` types and plutus types (types related to ``TxInfo``).
+
+* We will copy the ``Ledger.TimeSlot`` module in the emulator (ideally rename it) and keep it as
+  an internal module. Any functions not used by the emulator will be removed.
+
+* We will move the ``Ledger.Params`` module inside the emulator as an internal module and modify the
+  ``Params`` datatype name to ``EmulatorParams``.
+
+* We will modify the ``Plutus.Contract.Request.getParams`` function to
+  ``Plutus.Contract.Request.getProtocolParameters``. This implies to modify the name of Contract
```
Comment (kayvank) [inline doc/adr/0015-time-conversion.rst:46]:
```suggestion
  ``Plutus.Contract.Request.getProtocolParameters``. This implies modifying the name of Contract
```

**Review by sjoerdvisscher — APPROVED** [review]

File: doc/adr/0015-time-conversion.rst [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+The solution to create new effects in ``Plutus.Contract.Effects`` has the nice property that the
+implementation can differ depending on the environment.
+It allows us to keep using ``SlotConfig`` in the emulator, while using the existing implementation
+provided by ``ouroboros-consensus`` when using a real Cardano node.
+
+Implications
+------------
+
+* We will have to update the ``plutus-use-cases`` examples to use those new conversion functions.
+  The user will **not** use ``SlotConfig`` to work convert between slots and UTC time. He will
```
Comment (kayvank) [inline doc/adr/0015-time-conversion.rst:118]:
```suggestion
  The user will **not** use ``SlotConfig`` to convert between slots and UTC time. He will
```

**Review by kayvank — APPROVED** [review]

File: doc/adr/0015-time-conversion.rst (hunk 1 above)
Comment (berewt) [reply doc/adr/0015-time-conversion.rst:42]: Developpers may want to test how there contract behave with a given `SlotConfig`?

Comment (koslambrou) [reply doc/adr/0015-time-conversion.rst:42]: I think it *does* need to be configurable. The PAB simulator actually uses it to define the `delay` for creating new blocks.

**Review by berewt — APPROVED** [review]

File: doc/adr/0015-time-conversion.rst (hunk 1 above)
Comment (sjoerdvisscher) [reply doc/adr/0015-time-conversion.rst:42]: Ok!

**Pull request merged**